orchestrate-cockroachdb-with-kubernetes.md

```diff
@@ -0,0 +1,341 @@
+---
+title: Orchestrate CockroachDB with Kubernetes
+summary: 
+toc: false
+---
+
+[Kubernetes](http://kubernetes.io/) is an open-source system for automating the deployment, scaling, and management of containerized applications. Although Kubernetes is not yet production-ready for stateful applications such as CockroachDB, you can use Kubernetes' [`minikube`](https://github.com/kubernetes/minikube) tool to test out CockroachDB on Kubernetes locally, using persistent volumes and the alpha [Pet Set](http://kubernetes.io/docs/user-guide/petset/) feature to ensure persistent identity and storage for CockroachDB nodes. This page shows you how.
+
+<div id="toc"></div>
+
+## Step 1. Install and start Kubernetes locally
+
+Follow Kubernetes' [documentation](http://kubernetes.io/docs/getting-started-guides/minikube/) to install `minikube` and `kubectl` for your OS. Then start a local Kubernetes cluster:
+
+~~~ shell
+$ minikube start
+Starting local Kubernetes cluster...
+Kubectl is now configured to use the cluster.
+~~~
+
+## Step 2. Create persistent volumes and volume claims
+
+Each CockroachDB node will run in a [pod](http://kubernetes.io/docs/user-guide/pods/), which is a group of one or more Docker containers that can share resources. In our case, each pod will have only one container running a CockroachDB node. 
```

Comment:
> Due to running as a PetSet, the storage the containers will use is a little more complicated because we want it to be persistent and owned by a specific identity of a Cockroach node, which necessitates the use of persistent volumes and associated claims. Use the below script as a black-box for setting up the correct volumes and claims to make this example work.